### ISSUE TITLE
Better filtering for message handlers

### ISSUE THREAD

**Issue by jh0ker** [python-telegram-bot/python-telegram-bot#185]
Currently, everything that is not a command is matched by message handlers added with `Dispatcher.addTelegramMessageHandler`, including very different things like pictures, voice and status updates like group-joins etc.
I suggest having a range of filters that could be applied as either a whitelist or a blacklist. Example:

    dp.addTelegramMessageHandler(handler, filters=[AUDIO, VOICE], method=WHITELIST)
    dp.addTelegramMessageHandler(handler, filters=[STATUS_UPDATE], method=BLACKLIST)

**Comment by shelomentsevd** [comment]
@jh0ker, what dispatcher should do with messages, which are not in 'WHITELIST' and not in 'BLACKLIST'? 
How about something like this?
```
dp.addTelegramMessageHandler(audio_handler, filters=[AUDIO, VOICE])
dp.addTelegramMessageHandler(status_handler, filters=[STATUS_UPDATE])
```
Other messages will not be handled. Default value for filters is [AUDIO, VOICE, STATUS_UPDATE, TEXT, IMAGES, VIDEO].

**Comment by jh0ker** [comment]
So basically whitelist only?

**Comment by shelomentsevd** [comment]
@jh0ker 
Maybe filter "OTHER" for other messages?
`dp.addTelegramMessageHandler(other_handler, filters=[OTHER])`

**Comment by jh0ker** [comment]
I don't like that because it would make the handler dependent on other handlers, which I'm trying to avoid.

**Comment by jh0ker** [comment]
I have implemented the filtering now here: https://github.com/python-telegram-bot/python-telegram-bot/blob/dispatcher-rework/telegram/ext/messagehandler.py 
Check also the documentation on `Dispatcher.addHandler`: https://github.com/python-telegram-bot/python-telegram-bot/blob/dispatcher-rework/telegram/ext/dispatcher.py#L199

**Comment by leandrotoledo** [comment]
Resolving, this implementation is now released under 4.0rc1.

```
$ pip install python-telegram-bot==4.0rc1
```
